src/main/java/com/gary/backendv2/service/IncidentService.java

```diff
@@ -115,4 +150,28 @@ private Map<Integer, Incident> populateIncidentMap(List<Incident> incidents) {
 
 		return map;
 	}
+
+	private void assignDispatcher(Incident incident){
+		List<Dispatcher> dispatchers = dispatcherRepository.getAllByWorking(true);
+		if(dispatchers.size() == 0){
+			throw new HttpException(HttpStatus.NOT_ACCEPTABLE, "There is no dispatcher on duty");
+		}
+		List<Dispatcher> possibleAssigments = new ArrayList<>();
+		int activeIncidents = Integer.MAX_VALUE;
+		for (Dispatcher dispatcher:dispatchers) {
+			if(dispatcher.getOpenIncidents()<activeIncidents){
+				possibleAssigments.clear();
+				activeIncidents = dispatcher.getOpenIncidents();
+			}
+			if(dispatcher.getOpenIncidents() == activeIncidents){
+				possibleAssigments.add(dispatcher);
+			}
+		}
```

Comment:
nie rozumiem za bardzo logiki
czym jest openIncidents?
czemu iterujemy po wszystkich Dispatcherach z bazy i w sumie czemu w ogóle dyspozytor jest przypisywany do jakiegoś incydentu

Jeśli dobrze rozumiem to tutaj chcemy przypisać ratowników żeby gdzieś sobie pojechali na akcję. Jeśli tak to trzeba zdefiniować nowy typ pracownika który dziedziczy po AbstractEmployee. Bo Dispatcher to jest dyspozytor, nie ratownik

Reply:
Dyspozytora przypisujemy że się zajmuje jakimś incydentem a on przypisuje karetki nie ratowników. OpenIncidents daje liczbę obecnie otwartych incydentów by zawsze fostawal dyspozytor z najmniejsza ilością

Reply:
ok got it